code/datums/components/stationloving.dm

```diff
@@ -81,6 +81,9 @@
 		// Whenever shuttles move, everything seems to be on a hyperspace tile temporarily,
 		// so we need this to stop it from teleporting off of allowed shuttles.
 		if (istype(T, /turf/open/space/transit))
+			// We still might be on a disallowed shuttle,
+			// so we need to check again in a second to make sure.
+			addtimer(CALLBACK(src, PROC_REF(check_in_bounds)), 1 SECONDS)
```

Comment:
couldn't this potentially cause an infinite loop?

Reply:
@itsmeow If it ends up on a hyperspace turf outside of the initial shuttle z-level move, wouldn't the hyperspace turf just yeet it into deep space?

Reply:
@itsmeow just to ensure there's no infinite loops, I added an `is_retrying` parameter to `check_in_bounds` - it'll only do the retry once.